crates/ibc/src/core/ics03_connection/version.rs

```diff
@@ -95,29 +119,22 @@ pub fn get_compatible_versions() -> Vec<Version> {
     vec![Version::default()]
 }
 
-/// Selects a version from the intersection of locally supported and counterparty versions.
+/// Selects only one version from the intersection of locally supported and counterparty versions.
 pub fn pick_version(
     supported_versions: &[Version],
     counterparty_versions: &[Version],
 ) -> Result<Version, ConnectionError> {
-    let mut intersection: Vec<Version> = Vec::new();
-    for s in supported_versions.iter() {
-        for c in counterparty_versions.iter() {
-            if c.identifier != s.identifier {
-                continue;
-            }
-            for feature in c.features.iter() {
-                if feature.trim().is_empty() {
-                    return Err(ConnectionError::EmptyFeatures);
-                }
-            }
-            intersection.append(&mut vec![s.clone()]);
-        }
-    }
-    intersection.sort_by(|a, b| a.identifier.cmp(&b.identifier));
+    let mut intersection: Vec<Version> = counterparty_versions
+        .iter()
+        .filter(|v| v.verify_version_supported(supported_versions).is_ok())
+        .cloned()
+        .collect();
+
     if intersection.is_empty() {
         return Err(ConnectionError::NoCommonVersion);
     }
+
```

Comment:
ibc-go only returns the features that are contained in both the supported version, and the proposed version [here](https://github.com/cosmos/ibc-go/blob/295bab01f633633e0d6136ba9b974c0e248edbe5/modules/core/03-connection/types/version.go#L161)

Reply:
As far as I see, going through `filter(|v| v.verify_version_supported(supported_versions).is_ok())` ensure this check.

Reply:
[3d53f13](https://github.com/cosmos/ibc-rs/pull/626/commits/3d53f13a3a73dd23798ba317594f73664564991c)